src/capi/typeobject.cpp

```diff
@@ -1729,7 +1729,9 @@ extern "C" int PyType_Ready(PyTypeObject* cls) noexcept {
 
     RELEASE_ASSERT(cls->tp_getattro == NULL || cls->tp_getattro == PyObject_GenericGetAttr, "");
     RELEASE_ASSERT(cls->tp_setattro == NULL || cls->tp_setattro == PyObject_GenericSetAttr, "");
-    RELEASE_ASSERT(cls->tp_as_buffer == NULL, "");
+
+    // TODO: I think this is not implemented yet but the arraymodules uses it.
+    // RELEASE_ASSERT(cls->tp_as_buffer == NULL, "");
```

Comment:
Is calling bf_getreadbuffer and bf_getsegcount on an instance of an array, internally inside a modified pyston runtime function and checking that it's output enough or is there another way to trigger a buffer protocol read?

e.g.:
```C++
    Py_ssize_t totalBytes = 0;
    void* p = 0;
    Py_ssize_t numSegs =  arg->cls->tp_as_buffer->bf_getsegcount(arg, &totalBytes);
    Py_ssize_t numBytes = arg->cls->tp_as_buffer->bf_getreadbuffer(arg, 0, &p);
    printf("numBytes: %zd totalBytes: %zd numSegs %zd\n", numBytes, totalBytes, numSegs);
    for (int i=0; i<numBytes; ++i)
        printf("%c", ((char*)p)[i]);
```

This displays the correct output.